target/riscv/csr.c

```diff
@@ -323,7 +323,12 @@ static const target_ulong delegable_excps =
     (1ULL << (RISCV_EXCP_INST_GUEST_PAGE_FAULT)) |
     (1ULL << (RISCV_EXCP_LOAD_GUEST_ACCESS_FAULT)) |
     (1ULL << (RISCV_EXCP_STORE_GUEST_AMO_ACCESS_FAULT)) |
-    (1ULL << (RISCV_EXCP_CHERI));
+#ifdef TARGET_CHERI
+    (1ULL << (RISCV_EXCP_LOAD_CAP_PAGE_FAULT)) |
+    (1ULL << (RISCV_EXCP_STORE_AMO_CAP_PAGE_FAULT)) |
```

Comment:
We probably want to guard this with `#ifndef TARGET_RISCV32`, and maybe also the exceptions themselves (why is RISCV_EXCP_CHERI defined and delegable unconditionally?).

Reply:
I thought I fixed this to move the CHERI one under the #ifdef when it wasn't before?  I will try to make sure the exceptions have proper conditions in all places.

Reply:
I meant "was".